### PR TITLE
[ENH] Read annotation duration from SNIRF files.

### PR DESCRIPTION
**Pull request by rob-luke**
#### Reference issue
Before this PR, when reading SNIRF files, the annotation durations were ALL set to 1.0

#### What does this implement/fix?
This PR changes the `read_raw_snirf` function to read annotation durations from the file. 

#### Additional information
Relevant file type definition is here: https://github.com/fNIRS/snirf/blob/master/snirf_specification.md#nirsistimjdata

A PR has been opened to add correct stimulus duration to SNIRF writing at https://github.com/mne-tools/mne-nirs/pull/497

### CODE REVIEW

**Comment by rob-luke** [comment]
@agramfort @larsoner @drammock this PR is ready for review. There is a failing CI, but I think it's unrelated. Thanks!

File: mne/io/snirf/tests/test_snirf.py [hunk 1]
```diff
@@ -388,3 +390,18 @@ def test_annotation_description_from_stim_groups():
...
+    """Test annotation durations extracted correctly from stim group."""
+    raw = read_raw_snirf(snirf_nirsport2_20219, preload=True)
+    # Specify the expected SNIRF stim durations.
+    # We can verify these values should be 10 by using the official
+    # SNIRF package pysnirf2 and running the following script.
+    # You will see that the print statement shows the middle column,
+    # which represents duration, will be all 10s.
+    # from snirf import Snirf
+    # a = Snirf(snirf_nirsport2_20219, "r+"); print(a.nirs[0].stim[0].data)
+    expected_durations = np.ones((10)) * 10
```
Comment (larsoner) [inline mne/io/snirf/tests/test_snirf.py:406]:
```suggestion
    expected_durations = np.full((10,), 10)
```

**Review by larsoner — APPROVED** [review]
Pushed a couple of tiny commits, one to fix Azure and another to use `np.full` rather than np.ones(...) * 10), marking for merge when green -- thanks in advance @rob-luke !

File: mne/io/snirf/tests/test_snirf.py (hunk 1 above)
Comment (rob-luke) [reply mne/io/snirf/tests/test_snirf.py]: Oh nice, thanks @larsoner

**Pull request merged**